src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,228 @@
+package src.main.java.duke.parser;
+
+import static src.main.java.duke.commons.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static src.main.java.duke.commons.Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import src.main.java.duke.commands.AddDeadlineCommand;
+import src.main.java.duke.commands.AddEventCommand;
+import src.main.java.duke.commands.AddTodoCommand;
+import src.main.java.duke.commands.Command;
+import src.main.java.duke.commands.DeleteCommand;
+import src.main.java.duke.commands.ExitCommand;
+import src.main.java.duke.commands.FindCommand;
+import src.main.java.duke.commands.HelpCommand;
+import src.main.java.duke.commands.IncorrectCommand;
+import src.main.java.duke.commands.ListCommand;
+import src.main.java.duke.commands.MarkDoneCommand;
+
+/**
+ * Parses user input.
+ */
+public class Parser {
+
+    public static final Pattern TASK_INDEX_ARGS_FORMAT = Pattern.compile("(?<targetIndex>.+)");
+
+    public static final Pattern TASK_STRING_ARGS_FORMAT = Pattern.compile("(?<targetIndex>\\w.+)");
+
+    public static final Pattern EVENT_TASK_DATA_ARGS_FORMAT = // '/' forward slashes are reserved for delimiter prefixes
+            Pattern.compile("(?<description>[^/]+)+\\s+(\\/at)+\\s+"
+                    + "(?<duedate>[\\d\\d\\d-\\d\\d\\-\\d\\d\\d+\\s+\\d\\d:\\d\\d]+)");
+
+    public static final Pattern DEADLINE_TASK_DATA_ARGS_FORMAT =
+            Pattern.compile("(?<description>[^/]+)+\\s+(\\/by)+\\s+"
+                    + "(?<duedate>[\\d\\d\\d-\\d\\d\\-\\d\\d\\d+\\s+\\d\\d:\\d\\d]+)");
+
+    public static final Pattern TASK_DATA_ARGS_FORMAT = // '/' forward slashes are reserved for delimiter prefixes
+            Pattern.compile("(?<description>[^/]+)");
+
```

Comment:
I like how you have constant static variables to indicate the format for each task. You could perhaps consider changing the constant static variables to private instead as it appears that these constants are only used in Parser.java. Otherwise, nice job.

Reply:
Will implement this advice. Thank you.